### DISCUSSION TITLE
Experimental Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) 20230416

### DISCUSSION THREAD

**Discussion by brkirch** [brkirch/stable-diffusion-webui#8] [Announcements]
### Changelog

- Fix the `Maximum down sampling` setting for `Token Merging`

<hr /><em>This discussion was created from the release <a href='https://github.com/brkirch/stable-diffusion-webui/releases/tag/20230416_experimental'>Experimental Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) 20230416</a>.</em>

**Comment by chenzhonghua** [comment]
some extensions can not correct display, (e.g. Llul  https://github.com/hnmr293/sd-webui-llul.git)

**Comment by jrittvo** [comment]
In the issues at Llul  https://github.com/hnmr293/sd-webui-llul/issues/22  there are some people saying it needs Gradio 3.23.0.  This experimental build is using 3.16.2 because of other issues with the newer Gradio version.

**Comment by Vrk3ds** [comment]
I am getting this error. I have tried to upgrade, uninstall, re-install to no avail?

Traceback (most recent call last):
File "/Users/scottd/autosd/launch.py", line 382, in 
start()
File "/Users/scottd/autosd/launch.py", line 373, in start
import webui
File "/Users/scottd/autosd/webui.py", line 8, in 
from fastapi import FastAPI
ModuleNotFoundError: No module named 'fastapi'

Anyone else having this issue?

**Comment by jenxi** [comment]
Should I reinstall this again if I did a git pull update?

**Comment by bambooboo2020** [comment]
It always gives me an error after I generate the image multiple times, and then shuts down the server. Does anyone know how to solve this problem?

-[IOGPUMetalCommandBuffer validate]:215: failed assertion `commit an already committed command buffer'
Abort trap: 6
Mac-Studio:stable-diffusion-webui derek$ /Users/derek/stable-diffusion-webui/python/3.10.10/lib/python3.10/multiprocessing/resource_tracker.py:224: UserWarning: resource_tracker: There appear to be 1 leaked semaphore objects to clean up at shutdown
  warnings.warn('resource_tracker: There appear to be %d '

About my Mac Studio device:
GPU         Apple M1 Ultra(64 cores)
MacOS     13.3.1 (22E261)
Memory    128GB

**Reply by angstmann** [reply]
I get the exact same error very frequently.  Sometimes I can go a few generations no problem at all, sometimes it does it after every generation and I have to restart the WebUI. It's frustrating, as you never know right until the images is almost completed whether its going to crash or not.  Loving the speed and optimisations of this, but would love a little more stability on my M1 Pro MacBook Pro 16in.

**Reply by bambooboo2020** [reply]
I use the following command to start the service, which can reduce the frequency of crashes, but the problem still exists.
./webui.sh --no-half

**Reply by jenxi** [reply]
I’ve tried many options and this seems most stable on an M1 Max. 

`/webui.sh  --upcast-sampling --use-cpu interrogate --disable-nan-check --no-half-vae --opt-sub-quad-attention`

I updated to the latest Web UI 1.1.0 release and it is a lot more stable and I’m seeing significant improvements in generation speeds. You might want to consider switching to that? This installer seems outdated compared to the latest release.

**Reply by bambooboo2020** [reply]
> I’ve tried many options and this seems most stable on an M1 Max.
> 
> `/webui.sh --upcast-sampling --use-cpu interrogate --disable-nan-check --no-half-vae --opt-sub-quad-attention`
> 
> I updated to the latest Web UI 1.1.0 release and it is a lot more stable and I’m seeing significant improvements in generation speeds. You might want to consider switching to that? This installer seems outdated compared to the latest release.

Great job! It really works for me~ Thank you so much~

**Reply by geirdev** [reply]
> I’ve tried many options and this seems most stable on an M1 Max.
> 
> `/webui.sh --upcast-sampling --use-cpu interrogate --disable-nan-check --no-half-vae --opt-sub-quad-attention`
> 
> I updated to the latest Web UI 1.1.0 release and it is a lot more stable and I’m seeing significant improvements in generation speeds. You might want to consider switching to that? This installer seems outdated compared to the latest release.

You mean webui UI 1.1.0 from master git, not build for apple silicon?

**Reply by geirdev** [reply]
> > I’ve tried many options and this seems most stable on an M1 Max.
> > `/webui.sh --upcast-sampling --use-cpu interrogate --disable-nan-check --no-half-vae --opt-sub-quad-attention`
> > I updated to the latest Web UI 1.1.0 release and it is a lot more stable and I’m seeing significant improvements in generation speeds. You might want to consider switching to that? This installer seems outdated compared to the latest release.
> 
> You mean webui UI 1.1.0 from master git, not build for apple silicon?

Oh, I don't need the M1 only version anymore, thanks, I found out thanks to you.

**Reply by sidbnoida** [reply]
hi @undeadx1.....I am also really stuck here. I have also installed this "Experimental Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) 20230416" from here and am getting the same error. Do I need to reinstall webui from somewhere else or just run /webui.sh --upcast-sampling --use-cpu interrogate --disable-nan-check --no-half-vae --opt-sub-quad-attention ?

**Reply by geirdev** [reply]
> hi @undeadx1.....I am also really stuck here. I have also installed this "Experimental Offline Standalone Mac Installer for Stable Diffusion Web UI (unofficial) 20230416" from here and am getting the same error. Do I need to reinstall webui from somewhere else or just run /webui.sh --upcast-sampling --use-cpu interrogate --disable-nan-check --no-half-vae --opt-sub-quad-attention ?

hello.
I was able to successfully run it on m1 max by git cloning from this(https://github.com/AUTOMATIC1111/stable-diffusion-webui) repository and simply running it as ./webui.sh. The speed was not much different from m1 only webui.

Importantly, you will need to update to a pytorch 2.0 environment.
First make sure that pytorch 2.0 has been applied to python 3.10.
I think the webui for apple silicon is obsolete after pytorch 2.0.
